lib/bugsnag.js

```diff
@@ -92,7 +103,7 @@ Bugsnag.notify = function(error, options, cb) {
         options = {};
     }
     if (!Bugsnag.shouldNotify()) {
-        if (cb) {
+        if (cb && Configuration.logAllDeliveryErrors) {
```

Comment:
I think this branch could be in a more appropriate place, as the the option mentions `log` but no logging is done here (an `Error` object is created and passed to the `cb()`).

A better place would be in the definition of [`autoNotifyCallback`](https://github.com/bugsnag/bugsnag-node/blob/37c0c7bb67d551b5c1fdf57d3e324367a4f7e3d4/lib/bugsnag.js#L20-L25).